cmd/dep/main.go

```diff
@@ -312,3 +334,66 @@ func (c *commentWriter) Write(p []byte) (int, error) {
 	}
 	return len(p), nil
 }
+
+type profile struct {
+	cpuProfile string
+
+	memProfile     string
+	memProfileRate int
+
+	mutexProfile         string
+	mutexProfileFraction int
+
+	// TODO(jbd): Add block profile and -trace.
+
+	f *os.File // file to write the profiling output to
+}
+
+func (p *profile) start() error {
+	switch {
+	case p.cpuProfile != "":
+		if err := p.createOutput(p.cpuProfile); err != nil {
+			return err
+		}
+		return pprof.StartCPUProfile(p.f)
+	case p.memProfile != "":
+		if p.memProfileRate > 0 {
+			runtime.MemProfileRate = p.memProfileRate
+		}
+		return p.createOutput(p.memProfile)
+	case p.mutexProfile != "":
+		if p.mutexProfileFraction > 0 {
+			runtime.SetMutexProfileFraction(p.mutexProfileFraction)
+		}
+		return p.createOutput(p.mutexProfile)
+	}
```

Comment:
Would it be useful to support multiple profiles concurrently?  If not, perhaps specifying more than one should be an error or warning.